### PR TITLE
fix metadata sync

### PR DESCRIPTION
**Pull request by kinarashah**
1) metadata CRDs were only updated when `sendRKELabel` was true (if they were supposed to send to RKE). We now always update the content, so in case user wants to override the values, they can simply set the label true after refresh

2) labels comparison (causing multiple REST Updates in debug logs, but never updating actual object)
a) System Images: `"cattle.io/creator": "norman"` is present in our CRD labels and it wasn't considered for comparing object labels for update
b) Addons, Service Options: we check for a label (`sendRKELabel`)'s value, but if the label is not present, it should default to true 

3) added logs for easier debugging 

https://github.com/rancher/rancher/issues/24675

### CODE REVIEW

File: pkg/controllers/management/kontainerdrivermetadata/data.go [hunk 1]
```diff
@@ -314,7 +334,7 @@ func getLabelMap(k8sVersion string, data map[string]map[string]string,
 	if err != nil {
 		return nil, fmt.Errorf("k8sVersion not sem-ver %s %v", k8sVersion, err)
 	}
-	labelMap := map[string]string{}
+	labelMap := map[string]string{"cattle.io/creator": "norman"}
```
Comment (cjellick) [inline pkg/controllers/management/kontainerdrivermetadata/data.go]: Why do you need to do this?

**Comment by cjellick** [comment]
Commit messages should be more descriptive of the problem and the solution

File: pkg/controllers/management/kontainerdrivermetadata/data.go (hunk 1 above)
Comment (kinarashah) [reply pkg/controllers/management/kontainerdrivermetadata/data.go]: our crds always get this label, so if I don't consider it for comparison -- there's always need to update labels on sync

File: pkg/controllers/management/kontainerdrivermetadata/data.go [hunk 2]
```diff
@@ -206,15 +206,22 @@ func (md *MetadataController) createOrUpdateSystemImageCRD(k8sVersion string, sy
 			return err
 		}
 		return nil
-
 	}
-	if reflect.DeepEqual(sysImage.SystemImages, systemImages) && reflect.DeepEqual(sysImage.Labels, pluginsMap) {
+	dataEqual := reflect.DeepEqual(sysImage.SystemImages, systemImages)
```
Comment (cjellick) [inline pkg/controllers/management/kontainerdrivermetadata/data.go]: are these map or list comparisons (on my phone so hard to tell). If they are lists, deep equals is dangerous bc it takes into account order but we don’t really care about order of labels (do we?)

Comment (cjellick) [reply pkg/controllers/management/kontainerdrivermetadata/data.go]: I’m not sure on this one, just my reaction. So if you feel deepequals is safe in this (and the other below cases) respond with justification and we can move on

Comment (kinarashah) [reply pkg/controllers/management/kontainerdrivermetadata/data.go]: SystemImages is struct comparison with all fields string, so thought it should be ok? 

For labels, I use custom logic https://github.com/rancher/rancher/blob/b4ba60ce4408b75dda0f2f33a4ab5b0227d09c55/pkg/controllers/management/kontainerdrivermetadata/data.go#L688

Comment (cjellick) [reply pkg/controllers/management/kontainerdrivermetadata/data.go]: Ok

**Pull request merged**